### DISCUSSION TITLE
Alternative pages folder

### DISCUSSION THREAD

**Discussion by davestewart** [ElMassimo/iles#122] [Q&A] [answered]
Hey!

FYI, I got linked here from a Gridsome thread (having since moved to VuePress) but this looks like a great framework!

Anyway, question: would it be possible to have a project structure like this?

```
+- src
    +- app
    |   +- components
    |   |   +- ...
    |   +- layouts
    |   |   +- ...
    |   +- app.ts
    +- content           <-- this is what would have been /pages
        +- blog
        |   +- ...
        +- portfolio
        |   +- ...
        +- index.md
```

I would like to keep logic and content separate for various reasons: find and replace, commits, etc.

**Comment by ElMassimo** [comment, marked as answer]
Hi Dave! @davestewart 

You can [specify a different `pagesDir`](https://github.com/ElMassimo/iles/blob/main/packages/pages/src/types.ts#L68-L72), relative to the [`srcDir`](https://github.com/ElMassimo/iles/blob/main/packages/iles/types/shared.d.ts#L236-L240).

I haven't tested it when `pagesDir` is outside `srcDir`, but you could try with:

```ts
export default defineConfig({
  srcDir: 'src/app',
  pagesDir: '../content',
})
```

**Reply by ElMassimo** [reply]
Seems to work nicely, except for [auto-importing composables](https://github.com/ElMassimo/iles/blob/main/packages/iles/src/node/plugin/plugin.ts#L152).

Don't want to use a filesystem check for performance, but it could be fixed by also checking for `pagesDir`. A pull request is welcome, if you are interested.
